pkg/kubectl/resource/visitor.go

```diff
@@ -430,6 +335,62 @@ func (v FlattenListVisitor) Visit(fn VisitorFunc) error {
 	})
 }
 
+func ignoreFile(path string, extensions []string) bool {
+	if len(extensions) == 0 {
+		return false
+	}
+	ext := filepath.Ext(path)
+	for _, s := range extensions {
+		if s == ext {
+			return false
+		}
+	}
+	return true
+}
+
+// ExpandPathsToStreamVisitors will return a slice of Stream Visitors that will read and handle files from the provided path.
+// The returned visitor will be a StreamVisitor as reading a file can be the same as reading from STDIN. Paths argument also
+// accepts a single file, and will return a single visitor
+func ExpandPathsToStreamVisitors(mapper *Mapper, paths string, recursive bool, extensions []string, ignoreErrors bool, schema validation.Schema) ([]Visitor, error) {
+	var visitors []Visitor
+
+	err := filepath.Walk(paths, func(path string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if fi.IsDir() {
+			if path != paths && !recursive {
+				return filepath.SkipDir
+			}
+			return nil
+		}
+		if ignoreFile(path, extensions) {
+			return nil
+		}
+
+		f, err := os.Open(path)
+		if err != nil {
+			return fmt.Errorf("unable to open %q: %v", path, err)
+		}
+
```

Comment:
It's too easy to leave files open this way.  I'm concerned about any use of the Builder outside if a one-and-done command, which makes this leak file handles.  Can we solve this in a way that we don't leak?